Cosminis/Services/FriendServices.cs

```diff
@@ -0,0 +1,196 @@
+    {
+        try
+        {
+            Friends checkFriendship = _friendsRepo.FriendsByUserIds(searchingUserId, user2BeSearchedFor);
+            if(checkFriendship == null)
+            {
+                throw new ResourceNotFound();
+            }
+            return _friendsRepo.FriendsByUserIds(searchingUserId, user2BeSearchedFor);
+        }
+        catch (Exception E)
+        {
+            throw;
+        }
+    }
+
+    public Friends EditFriendship(int editingUserID, int user2BeEdited, string status)
+    {
+        try
+        {
+            Friends editFriendship = _friendsRepo.EditFriendship(editingUserID, user2BeEdited, status);
+            if(editFriendship == null)
+            {
+                throw new ResourceNotFound();
+            }
+            return _friendsRepo.EditFriendship(editingUserID, user2BeEdited, status);
+        }
+        catch (Exception E)
+        {
+            throw;
+        }
+    }    
+
+    public List<Friends> CheckRelationshipStatusByUserId(int searchingId, string status)
+    {
+        try
+        {
+            List<Friends> checkRelationship = _friendsRepo.CheckRelationshipStatusByUserId(searchingId, status);
+            if(checkRelationship == null)
+            {
+                throw new ResourceNotFound("No relationship with this status exists.");
+            }
+            return _friendsRepo.CheckRelationshipStatusByUserId(searchingId, status);
+        }
+        catch (Exception E)
+        {
+            throw;
+        }
+    }            
+
+    public List<Friends> CheckRelationshipStatusByUsername(string username, string status)
+    {
+        try
+        {
+            List<Friends> checkRelationship = _friendsRepo.CheckRelationshipStatusByUsername(username, status);
+            if(checkRelationship == null)
+            {
+                throw new ResourceNotFound("No relationship with this status exists.");
+            }
+            return _friendsRepo.CheckRelationshipStatusByUsername(username, status);
+        }
+        catch (Exception E)
+        {
+            throw;
+        }
+    } 
+
+    public List<Friends> ViewAllFriends(int userIdToLookup)
+    {
+        try
+        {
+            List<Friends> checkRelationship = _friendsRepo.ViewAllFriends(userIdToLookup);
+            if(checkRelationship == null)
+            {
+                throw new ResourceNotFound("This user has no friends.");
+            }
+            return _friendsRepo.ViewAllFriends(userIdToLookup);
+        }
+        catch (Exception E)
+        {
+            throw;
+        }
+    }   
+
+    public Friends AddFriendByUserId(int requesterId, int addedId)
+    {
+        User user2Add2 = new User();
+        user2Add2.UserId = addedId;
+        try
+        {
+            Friends checkRelationship = _friendsRepo.AddFriendByUserId(requesterId, addedId);
+            
+            int goldToAdd = 5;
+
+            _ResourceRepo.AddGold(user2Add2, goldToAdd);
```

Comment:
I love how you figured this one out without asking me to change ResourceRepo methods

Reply:
lol